### PR TITLE
feat: Add badges to reflect 3.12 support

### PR DESCRIPTION
**Pull request by reneleonhardt**
Thank your for contributing to Nuitka!

!! Please check that you select the **develop branch** (details see below link) !!

Before submitting a PR, please review the guidelines:
[Contributing Guidelines](https://github.com/Nuitka/Nuitka/blob/develop/CONTRIBUTING.md)

# What does this PR do?

* Add badge to reflect Python support [Python](https://badge.fury.io/py/Nuitka)
* Add PyPI badge [PyPI](https://badge.fury.io/py/Nuitka)
* Add CII Best Practices badge [CII Best Practices](https://www.bestpractices.dev/en/projects/8158)
* Add OpenSSF Scorecard badge [OpenSSF Scorecard](https://securityscorecards.dev/viewer/?uri=github.com/Nuitka/Nuitka)
* Add Contributor Covenant badge [Contributor Covenant](CODE_OF_CONDUCT.md)
* Add Python 3.12 support to setup and README
* Updated all occurences of 3.11 where possible
* Extra commit: Don't use 3.12 or skip test suite if requested

# Preview in Markdown
[Python](https://badge.fury.io/py/Nuitka)
[PyPI](https://badge.fury.io/py/Nuitka)
[CII Best Practices](https://www.bestpractices.dev/en/projects/8158)
[OpenSSF Scorecard](https://securityscorecards.dev/viewer/?uri=github.com/Nuitka/Nuitka)
[Contributor Covenant](CODE_OF_CONDUCT.md)

# Why was it initiated? Any relevant Issues?

I was celebrating the fresh 3.12 support and saw that the project wasn't showing proud Python badges like others do 😄

# PR Checklist

- [x] Correct base branch selected? Should be `develop` branch.
- [x] Enabled commit hook or executed `./bin/autoformat-nuitka-source`.
- [x] All tests still pass. Check the Developer Manual about `Running the Tests`. There are GitHub
  Actions tests that cover the most important things however, and you are welcome to rely on those,
  but they might not cover enough.
- [x] Ideally new features or fixed regressions ought to be covered via new tests.
- [x] Ideally new or changed features have documentation updates.

### CODE REVIEW

**Comment by kayhayen** [comment]
Reminds me, we probably don't announce 3.12 support in a lot of places yet.

File: Developer_Manual.rst [hunk 1]
```diff
@@ -828,59 +834,59 @@ Added new CPython suites
...
-for adding CPython311 based on the CPython310 branch.
+for adding CPython313 based on the CPython312 branch.
 
 .. code:: bash
 
    # Switch to a new branch.
-   git checkout CPython310
-   git branch CPython311
-   git checkout CPython311
+   git checkout CPython312
```
Comment (kayhayen) [inline Developer_Manual.rst:842]: Actually there is no CPython312 branch yet, this still needs to be completed for 311, so this part of the change is not yet due.

**Review by kayhayen — COMMENTED** [review]
Amazing job, thanks for doing this!

**Comment by kayhayen** [comment]
I have yet to resolve website vs. included documentation, the README.rst becoming more and obsolete, but the new README indeed should have all those badges and things.

File: Developer_Manual.rst (hunk 1 above)
Comment (reneleonhardt) [reply Developer_Manual.rst:842]: I actually checked beforehand, true 😅
Should I revert this file or is there no "danger" that developers try and fail until missing branches have been added?

**Comment by reneleonhardt** [comment]
> Amazing job, thanks for doing this!

Thank you very much, feels good if contributions are appreciated!
If you like badges, you probably will like the cool Dockerfile I prepared for you too... 😉
Spoiler alert:
```
docker run --rm --mount type=bind,source="${PWD}",target=/app nuitka "pip install click && nuitka --follow-imports --lto=yes --onefile trivyreport.py" && mv trivyreport.bin trivyreport
```
Hehe, I'm actually using the Dockerfile version of that so I can build my own multi-platform images from your image, but both possibilities are working 😄

I just couldn't figure out to exclude all files from the build folder (my app seems to need nothing except Python and math.so in trivyreport.app/Contents/MacOS so to speak, would save 10 MB of 24 MB uncompressed)...

**Comment by kayhayen** [comment]
You should checkout `bin/run-inside-nuitka-container` which I designed for transparent container usage, although on macOS I have had difficulties with cross architecture builds and docker, or was it podman, or both.

File: Developer_Manual.rst (hunk 1 above)
Comment (kayhayen) [reply Developer_Manual.rst:842]: I will be surprised if anybody but me ever executes the said procedure, sadly. So I guess, I can merge like that. As it stands, it's only example commands anyway.

**Comment by reneleonhardt** [comment]
> You should checkout `bin/run-inside-nuitka-container` which I designed for transparent container usage, although on macOS I have had difficulties with cross architecture builds and docker, or was it podman, or both.

Cool, didn't see that, will do 😄

**Pull request merged**